### ISSUE TITLE
Remove all compilation warnings when compiling binary

### ISSUE THREAD

**Issue by kolmodin** [kolmodin/binary#82]
From @simonpj:

In the binary library I’m seeing lots of these warnings:

```
libraries/binary/src/Data/Binary/Get.hs:420:1: warning:

    Rule "getWord16le/readN" may never fire

      because ‘getWord16le’ might inline first

    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on this function

libraries/binary/src/Data/Binary/Builder/Base.hs:510:1: warning:

    Rule "flush/flush" may never fire

      because ‘flush’ might inline first

    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on this function
```

The warnings look right to me: currently everything is very fragile and may not work as you intend.

**Comment by kolmodin** [comment]
Last warning will be fixed with #99.

**Comment by kolmodin** [comment]
Done! Wohoo!
